Unable to insert negatif floats

Hi, 

With the rc33, I'm getting a parse error whatever I try to insert a negatif float value into my db.

Here's a reproduction of the bug:

    curl -XPOST 'http://localhost:8086/write' -d '
    {
        "database": "mydb",
        "retentionPolicy": "default",
        "points": [
            {
                "measurement": "cpu_load_short",
                "tags": {
                    "host": "server01",
                    "region": "us-west"
                },
                "time": "2009-11-10T23:00:00Z",
                "fields": {
                    "value": -0.64
                }
            }
        ]
    }
    '

and I get : 

    unable to parse number value '-0.64.0': strconv.ParseFloat: parsing "-0.64.0": invalid syntax
    unable to parse number value '-0.64.0': strconv.ParseFloat: parsing "-0.64.0": invalid syntax

I have the same issue:

```
panic: unable to parse number value '-107.5.0': strconv.ParseFloat: parsing "-107.5.0": invalid syntax
```

InfluxDB crashes then.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,7 +6,7 @@
 - [#2931](https://github.com/influxdb/influxdb/pull/2931): Services and reporting should wait until cluster has leader.
 - [#2943](https://github.com/influxdb/influxdb/issues/2943): Ensure default retention policies are fully replicated
 - [#2948](https://github.com/influxdb/influxdb/issues/2948): Field mismatch error message to include measurement name
-
+- [#2919](https://github.com/influxdb/influxdb/issues/2919): Unable to insert negative floats
 
 ## v0.9.0 [2015-06-11]
 

--- a/tsdb/points.go
+++ b/tsdb/points.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"fmt"
 	"hash/fnv"
+	"math"
 	"sort"
 	"strconv"
 	"strings"
@@ -928,7 +929,8 @@ func (p Fields) MarshalBinary() []byte {
 			// ensure there is a decimal in the encoded for
 
 			val := []byte(strconv.FormatFloat(t, 'f', -1, 64))
-			hasDecimal := t-float64(int64(t)) > 0
+			_, frac := math.Modf(t)
+			hasDecimal := frac != 0
 			b = append(b, val...)
 			if !hasDecimal {
 				b = append(b, []byte(".0")...)

--- a/tsdb/points_test.go
+++ b/tsdb/points_test.go
@@ -522,6 +522,30 @@ func TestParsePointUnicodeString(t *testing.T) {
 	)
 }
 
+func TestNewPointNegativeFloat(t *testing.T) {
+	test(t, `cpu value=-0.64 1000000000`,
+		NewPoint(
+			"cpu",
+			Tags{},
+			Fields{
+				"value": -0.64,
+			},
+			time.Unix(1, 0)),
+	)
+}
+
+func TestNewPointFloatNoDecimal(t *testing.T) {
+	test(t, `cpu value=1. 1000000000`,
+		NewPoint(
+			"cpu",
+			Tags{},
+			Fields{
+				"value": 1.0,
+			},
+			time.Unix(1, 0)),
+	)
+}
+
 func TestParsePointIntsFloats(t *testing.T) {
 	pts, err := ParsePoints([]byte(`cpu,host=serverA,region=us-east int=10,float=11.0,float2=12.1 1000000000`))
 	if err != nil {
@@ -736,5 +760,4 @@ func TestNewPointEscaped(t *testing.T) {
 	if exp := `cpu\=main,tag\=bar=value\=foo name\=bar=1.0 0`; pt.String() != exp {
 		t.Errorf("NewPoint().String() mismatch.\ngot %v\nexp %v", pt.String(), exp)
 	}
-
 }

